app/controllers/users_controller.rb

```diff
@@ -37,13 +37,19 @@ def update
     end
   end
 
-  def destroy
-    @user = User.find(params[:id])
-    authorize! :destroy, @user
+  def confirm_destroy
```

Comment:
Why do you not use Devise::Models::DatabaseAuthenticatable#destroy_with_password from the devise library? http://www.rubydoc.info/github/plataformatec/devise/Devise/Models/DatabaseAuthenticatable:destroy_with_password

Reply:
Right you are